hpack/hpack.py

```diff
@@ -88,21 +89,19 @@ def decode_integer(data, prefix_bits):
     number of bytes that were consumed from ``data`` in order to get that
     integer.
     """
-    if not (0 < prefix_bits < 9):
+    if prefix_bits < 1 or prefix_bits > 8:
         raise ValueError(
-            "Prefix bits must be between 0 and 9, got %s" % prefix_bits
+            "Prefix bits must be between 1 and 8, got %s" % prefix_bits
         )
 
-    max_number = (2 ** prefix_bits) - 1
-    mask = 0xFF >> (8 - prefix_bits)
+    max_number = _PREFIX_BIT_MAX_NUMBERS[prefix_bits]
     index = 0
     shift = 0
 
     try:
-        number = to_byte(data[index]) & mask
+        number = to_byte(data[index]) & (0xFF >> (8 - prefix_bits))
```

Comment:
Saves an access as it's only used in the one place. I will revert this change for readability.